app/helpers/room_bookings_helper.rb

```diff
@@ -9,4 +9,103 @@ def available_courses(schedule)
                      .where(room_bookings: { id: nil })
     render partial: '/shared/courses_list', locals: { courses: @courses }
   end
+
+  def eligible_instructors(schedule, booking)
+    time_slot = TimeSlot.find_by(id: booking.time_slot_id)
+    current_instructor_id = booking.instructor_id
+    course_id = booking.course.id
+    before_9 = Time.parse('09:00 AM')
+    after_3 = Time.parse('03:00 PM')
+    return [] if schedule.instructors.nil?
+
+    ans = schedule.instructors&.select do |instructor| # which instructors have time
```

Comment:
Ideally we would want to do this part as a complex query rather than fetching instructors and then filtering but since our fetches are already taking forever I think this should be a good intermediate fix for now.